plugins/env/app/models/environment_variable.rb

```diff
@@ -78,7 +78,7 @@ def env_vars_from_external_groups(project, deploy_group)
         all_groups = Rails.cache.fetch("env-#{group.url}-read", expires_in: EXTERNAL_VARIABLE_CACHE_DURATION) do
           group.read
         end
-        all_groups[deploy_group.permalink]
+        all_groups[deploy_group.permalink] || all_groups["*"]
```

Comment:
app/models/environment.rb uses `[["All", nil]]` so would be nice if we can use any of those